### PR TITLE
v1.0.2

### PR DESCRIPTION
**Pull request by oranngennew**
Version 1.0.2 
- [x] Args in "widget-detected" post message
- [x] Lightbox in modal mode
- [x] Target for mobile version
- [x] Auto adjust size
- [x] Optional hide preloader

### CODE REVIEW

File: src/childwindow.js [hunk 1]
```diff
@@ -50,6 +43,9 @@ module.exports = (function () {
             this.triggerEvent('load');
             $(event.target).off(event);
         }, this));
+        message.on('widget-detection', function () {
```
Comment (andreysalnikov) [inline src/childwindow.js]: Child window has't been created at this moment, so you can't set event listener for it

File: src/lightbox.js [hunk 2]
```diff
@@ -27,10 +31,10 @@ module.exports = (function () {
     var TEMPLATE = '<div class="<%-prefix%>">' +
         '<div class="<%-prefix%>-overlay"></div>' +
         '<div class="<%-prefix%>-content <%-prefix%>-content__hidden">' +
-            '<iframe class="<%-prefix%>-content-iframe" src="<%-url%>" frameborder="0"></iframe>' +
```
Comment (andreysalnikov) [inline src/lightbox.js]: Give html formatting

File: src/lightbox.js [hunk 3]
```diff
@@ -61,20 +65,29 @@ module.exports = (function () {
...
 
     /** Public Members **/
-    LightBox.prototype.openFrame = function(url, options) {
-        options = _.extend({}, DEFAULT_OPTIONS, options);
+    LightBox.prototype.openFrame = function (url, options) {
+        this.options = _.extend({}, this.options, options);
+        options = this.options;
 
         var bodyElement = $(global.document.body);
+        var spinner = options.spinner !== 'none' ?
```
Comment (andreysalnikov) [inline src/lightbox.js]: Add empty value with key 'none' into SPINNERS, and remove this condition

File: src/lightbox.js [hunk 4]
```diff
@@ -179,9 +190,10 @@ module.exports = (function () {
 
         var loadTimer;
         lightBoxIframeElement.on('load', _.bind(function (event) {
+            var timeout = options.autoSize ? 30000 : 1000; //30000 if psDimensions will not arrive
```
Comment (andreysalnikov) [inline src/lightbox.js]: 30 sec is too much

File: README.md [hunk 5]
```diff
@@ -94,7 +94,7 @@ define(['PATH_TO_WIDGET/embed'], function (XPayStationWidget) {
     * **closeByKeyboard** — Toggle if pressing of ESC key should close lightbox, default true
     * **contentBackground** — Background of the frame, default is '#ffffff'
     * **contentMargin** — margin around frame, default '10px',
-    * **spinner** — Type of animated loading spinner, can be 'xsolla' or round, default is the first one
```
Comment (andreysalnikov) [inline README.md]: Describe **modal** and **autoSize** options

Comment (oranngennew) [inline README.md]: done

File: src/lightbox.js (hunk 3 above)
Comment (oranngennew) [inline src/lightbox.js]: done

File: src/lightbox.js (hunk 2 above)
Comment (oranngennew) [inline src/lightbox.js]: Given!

**Pull request merged**